xarray/core/variable.py

```diff
@@ -216,7 +217,16 @@ def _possibly_convert_objects(values):
     * pd.Timedelta
     """
     as_series = pd.Series(values.ravel(), copy=False)
+    # For why we need this behavior: https://github.com/pandas-dev/pandas/issues/61938
```

Comment:
let's add a little more please "pandas will convert to str if possible; preserve object dtype metadata in all other cases"